source/public/scripts/main.journal.js

```diff
@@ -0,0 +1,127 @@
+      var tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+
+      if (editingIndex >= 0) {
+        // If editing an existing task, update its details
+        tasks[editingIndex] = {
+          name: taskDesc,
+          type_tag: taskType,
+          project_tag: taskProject,
+          completed: tasks[editingIndex].completed // Preserve the completed status
+        };
+      } else {
+        // If adding a new task, create a new task object and add to the tasks array
+        var task = {
+          name: taskDesc,
+          type_tag: taskType,
+          project_tag: taskProject,
+          completed: false // New tasks are not completed by default
+        };
+        tasks.push(task);
+      }
+
+      // Save the updated tasks array to local storage
+      localStorage.setItem("tasks", JSON.stringify(tasks));
+
+      // Refresh the displayed tasks and hide the modal
+      displayTasks();
+      modal.style.display = "none";
+    }
+
+    // Hide the modal when the "Cancel" button is clicked
+    cancelBtn.onclick = function () {
+      modal.style.display = "none";
+    }
+
+    // Delete the current task when the "Delete" button in the modal is clicked
+    deleteBtnModal.onclick = function () {
+      var tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+      if (editingIndex >= 0) {
+        // Remove the task at the editingIndex from the tasks array
+        tasks.splice(editingIndex, 1);
+        // Save the updated tasks array to local storage
+        localStorage.setItem("tasks", JSON.stringify(tasks));
+        // Refresh the displayed tasks and hide the modal
+        displayTasks();
+        modal.style.display = "none";
+      }
+    }
+
+    // Function to display tasks from local storage
+    function displayTasks() {
+      var tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+      taskContainer.innerHTML = ""; // Clear the current task display
+      tasks.forEach(function (task, index) {
+        // Create a new task item element
+        var taskElement = document.createElement("div");
+        taskElement.className = "task-item";
+        taskElement.innerHTML = `
+          <input type="checkbox" id="task-${index}" ${task.completed ? 'checked' : ''}>
+          <label for="task-${index}">Task ${index + 1}: ${task.name}, Type: ${task.type_tag}, Project: ${task.project_tag}</label>
+          <button class="edit-task" data-index="${index}">Edit</button>
+        `;
+        taskContainer.appendChild(taskElement); // Append the new task item to the container
+
+        // Add event listener to the checkbox to update the task's completed status
+        var checkbox = taskElement.querySelector(`input[type="checkbox"]`);
+        checkbox.addEventListener('change', function () {
+          tasks[index].completed = this.checked; // Update completed status
+          localStorage.setItem("tasks", JSON.stringify(tasks)); // Save updated tasks array to local storage
```

Comment:
Since this line is being called a lot, maybe for readability turning it into a small function like `updateTasks()` would be helpful?